### PR TITLE
fix: chat window scroll

### LINKED ISSUE #3: Chat window doesn't scroll to the bottom automatically
## What

When you enter a chat room the conversation window doesn't automatically scroll and start from the bottom

## Why

So users can see the latest /recent message

image

### PR DESCRIPTION
**Pull request by jason1985**
# Description
This PR fixes #3
# Demo
demo
# Thanks
Let me know if I need to change anything or if this works for you.

Thank you!

### CODE REVIEW

File: src/components/chatroom/Chatroom.tsx [hunk 1]
```diff
@@ -32,6 +32,10 @@ export const Chatroom = (props: {
   const [formMessage, setFormMessage] = useState('')
   const { id } = useParams<{ id: string }>()
 
+  useEffect(()=> {
+    scrollToBottom()
+  },[messages])
+
```
Comment (benji011) [inline src/components/chatroom/Chatroom.tsx:38]: Very nice use of the `useEffect` function!

**Comment by benji011** [comment]
This is awesome! Very simple solution and it works fine on first render. 

I did see that the issue still persists when moving from 1 chatroom to the next. My guess is that the below function when called on click isn't taking into account the height of the window because the number of threads maybe isn't loaded

https://github.com/benji011/yggdrasil/blob/6b77c65e627159bd41154dd43c756145e768ee21/src/components/header/Threads.tsx#L37-L41

Maybe we can try setting

`chatroomWindowElement.scrollTop` in it's own state, then pass it in as a prop

**Comment by jason1985** [comment]
Thank you!

I'm not very familiar with threads and not exactly sure how to go about setting `chatroomWindowElement.scrollTop` in it's own state.

However I just tried this and it seems to be working ok.
```js
  const goToChatroom = async (thread: IThread) => {
    await setThreadData(thread)
    await setShowThreads(!showThreads)
    scrollToBottom()
  }
```

**Comment by benji011** [comment]
That will actually not compile because the `await` keyword is reserved for async functions. e.g. when you are returning a new Promise. [This doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) describes in great detail how you can take advantage of this mechanism.

I realise this is somewhat beyond the scope of this issue so I will branch it out as a separate task (and if you're interested by all means have a crack at it!) 

You did a really good job here so it definitely deserves my approval :)

**Review by benji011 — APPROVED** [review]

**Comment by jason1985** [comment]
Thank you very much!
I appreciate you letting me work on this.

**Pull request merged**